packages/backend/src/tokens/interstitial.ts

```diff
@@ -63,6 +66,7 @@ export function loadInterstitialFromLocal(options: Omit<LoadInterstitialOptions,
                 try {
                     await Clerk.load({
                         isSatellite: ${isSatellite},
+                        shouldSyncLink: ${debugData?.reason !== 'satellite-cookie-missing'},
```

Comment:
This is interesting because you're doing `!== 'satellite-cookie-missing'` instead of `=== satellite-uat-missing` (which I suggested renaming `satellite-needs-sync`).

The result is that this will sync/link on every interstitial, including when the JWT has expired but the session may still be active.

This is a divergence from what we do for CSR apps.  In CSR, if a JWT expires we simply retrieve a new one, we do not resync.

Which behavior is more desirable?  
1. If we resync on expired JWT, that means a user can manually clear their cookies from primary and we'll detect it on the next visit.
2. If we don't resync on expired JWT, that means we won't detect signs outs by way of manually cleared cookies. Instead, the user will need to click a sign out button.  Also, it means there won't be a sync flow while the user is signed in.

Can we start with (2) just to have a match with CSR to start?  We can run this by some customers to figure out which is more desirable.

```suggestion
                        shouldSyncLink: ${isSatellite && debugData?.reason === 'satellite-needs-sync'},
```

Reply:
@colinclerk I am very skeptic about these suggested changes and I don't think it will work.

Bellow `this.isSatellite` is a getter that inside handles the logic of `this.#options.isSatellite` beeing a function

In case of isSatellite is a function the above check `isSatellite && debugData?.reason === 'satellite-needs-sync'` is not correct.

Also `satellite-cookie-missing` is different than `satellite-uat-missing` which you proposed renaming to `satellite-needs-sync`


In the interstitial rules if `satellite-cookie-missing` is thrown that means we should not sync & link again as we already have a valid `client_uat`. All that need to be done is for the interstitial to reload in order to get the new `__session` cookie

Reply:
What I understand from this is this we need to reverse the current logic of "Always sync except when ..."
to "Only Sync if ...."

Also I want to point out the this logic  `shouldSyncLink: ${isSatellite && debugData?.reason === 'satellite-needs-sync'}` would only work in interstitial as the `isSatellite` in @clerk/backend is boolean, where as in remix, nextjs & clerk-js it could also be a function.

Does that mean that the sync/link should ONLY be triggered by the interstitial ?

Reply:
@colinclerk May we review this in a following PR as we need some of these changes to proceed with testing E2E flows on production ?
If I've understood correctly the remaining suggestions are related to improvements

Reply:
yes we can review later

Reply:
Please make the first suggestion, at least. The first one is just for readability